programs/ssl/ssl_server2.c

```diff
@@ -2564,23 +2597,33 @@ int main( int argc, char *argv[] )
 #if defined(MBEDTLS_USE_PSA_CRYPTO)
     if( opt.key_opaque != 0 )
     {
-        psa_algorithm_t psa_alg, psa_alg2;
-        psa_key_usage_t psa_usage;
+        psa_algorithm_t psa_alg, psa_alg2 = 0;
+        psa_key_usage_t psa_usage = 0;
 
-        if ( mbedtls_pk_get_type( &pkey ) == MBEDTLS_PK_ECKEY ||
-             mbedtls_pk_get_type( &pkey ) == MBEDTLS_PK_RSA )
+        if( strcmp( opt.key1_opaque_alg1, DFL_KEY_OPAQUE_ALG ) == 0 )
         {
             if( mbedtls_pk_get_type( &pkey ) == MBEDTLS_PK_ECKEY )
             {
-                psa_alg = PSA_ALG_ECDSA( PSA_ALG_ANY_HASH );
-                psa_alg2 = PSA_ALG_ECDH;
-                psa_usage = PSA_KEY_USAGE_SIGN_HASH | PSA_KEY_USAGE_DERIVE;
+                opt.key1_opaque_alg1 = "ecdsa-sign";
+                opt.key1_opaque_alg2 = "ecdh";
             }
-            else
+            else if( mbedtls_pk_get_type( &pkey ) == MBEDTLS_PK_RSA )
             {
-                psa_alg = PSA_ALG_RSA_PKCS1V15_SIGN( PSA_ALG_ANY_HASH );
-                psa_alg2 = PSA_ALG_NONE;
-                psa_usage = PSA_KEY_USAGE_SIGN_HASH;
+                opt.key1_opaque_alg1 = "rsa-sign-pkcs1";
+                opt.key1_opaque_alg2 = "none";
```

Comment:
Hmm, why set this to "none" while client-side rsa-sign-pss is allowed?

Reply:
Because in original code we had:
`psa_alg2 = PSA_ALG_NONE;`
so to achieve the same settings `opt.key1_opaque_alg2` must be "none".